mdetsims/sim_utils.py

```diff
@@ -607,9 +607,9 @@ def _get_band_objects(self):
         positions = []
 
         nobj = self._get_nobj()
+
         # turn on/off varying shear
-        if self.flag:
-            shear_red = self.sel.choice(2,nobj,p=[1-self.fraction,self.fraction])
+        shear_red = self.sel.choice(2, nobj, p=[self.frac_ex, 1-self.frac_ex])
```

Comment:
I remove the if statement. From now on, we don't need to worry about any branching issue here.